docs/sql-programming-guide.md

```diff
@@ -1073,6 +1073,16 @@ the following case-sensitive options:
   </tr>
 
   <tr>
+     <td><code>numPartitions</code></td>
+     <td>
+       The number of partitions that can be used, if set. It works by limiting both read and write
+       operations' parallelism. If the number of partitions to write exceeds this limit, the
+       operation will coalesce the data set to fewer partitions before writing. In other words,
```

Comment:
Conceptually, if the column values are exactly identical, only one partition will be generated. 

The bug in https://github.com/apache/spark/pull/15499 is not related the above fact.